libpromises/evalfunction.c

```diff
@@ -8722,6 +8722,35 @@ static FnCallResult FnCallCFEngineCallers(EvalContext *ctx, ARG_UNUSED const Pol
     return FnReturnContainerNoCopy(callers);
 }
 
+static FnCallResult FnCallString(ARG_UNUSED EvalContext *ctx,
+                                 ARG_UNUSED const Policy *policy,
+                                 ARG_UNUSED const FnCall *fp,
+                                 const Rlist *finalargs)
+{
+    // Parser doesnt allow this to be true since STRING_ARGS has one argument
+    if(finalargs == NULL)
+    {
+        return FnFailure();
+    }
+
+    char *ret;
+    bool allocated = false;  // Does it imply heap allocation is present?
+    JsonElement* json = VarNameOrInlineToJson(ctx, fp, finalargs, false, &allocated);
+
+    if(json != NULL)
+    {
+        Writer* w = StringWriter();
+        JsonWriteCompact(w, json);
+        ret = StringWriterClose(w);
```

Comment:
Looks like there should be `JsonDestroy()` somewhere here.

Reply:
Is the JsonDestroy necessary if [allocated](https://github.com/cfengine/core/blob/b248be4d04c9c508f6b76fe9b24fe40be9658aad/libpromises/evalfunction.c#L495) is set to true by the function? There are nested calls that push that pointer to bool further down but i cant find documentation on it.

Reply:
Yes, `allocated == true` means the `json` was allocated and either `JsonDestroy()` is needed after it is used/converted or it needs to be returned to the caller (not this case).